test/Microsoft.AspNet.JsonPatch.Test/NestedObjectTests.cs

```diff
@@ -202,6 +202,151 @@ public void AddToListWithSerialization()
         }
 
         [Fact]
+        public void AddToIntListToGeneralList()
```

Comment:
Do all of these tests vary on the declared type of the property? (Didn't you update the code to use the runtime type).

I'd also like to see one where we're modifying a list through a nested property.